pkg/framework/util.go

```diff
@@ -297,9 +298,21 @@ func ListImage(c internalapi.ImageManagerService, filter *runtimeapi.ImageFilter
 
 // PullPublicImage pulls the public image named imageName.
 func PullPublicImage(c internalapi.ImageManagerService, imageName string, podConfig *runtimeapi.PodSandboxConfig) string {
-	if !strings.Contains(imageName, ":") {
-		imageName = imageName + ":latest"
-		Logf("Use latest as default image tag.")
+
+	ref, err := reference.ParseNamed(imageName)
```

Comment:
report an error when `err != nil`? In what cases, the error would happen?

Reply:
Hm, yeah I guess we have to handle three possible paths here:

- The image name is fully qualified, then we parse it and modify the domain
- The image name is not fully qualified, then we can prepend the registry-prefix
- The image name is wrong, then we fail.

I adapted the implementation to work like this.

Reply:
thanks